controller/instance_manager_controller.go

```diff
@@ -547,15 +543,21 @@ func (imc *InstanceManagerController) syncLogSettingsToInstanceManagerPod(im *lo
 		}
 
 		switch settingName {
-		case types.SettingNameV2DataEngineLogLevel:
-			err = client.LogSetLevel(longhorn.DataEngineTypeV2, "spdk_tgt", setting.Value)
+		case types.SettingNameLogLevel:
+			if types.IsDataEngineV1(im.Spec.DataEngine) {
+				err = client.LogSetLevel(longhorn.DataEngineTypeV1, "", setting.Value)
+			} else {
+				err = client.LogSetLevel(longhorn.DataEngineTypeV2, "", setting.Value)
+			}
 			if err != nil {
-				return errors.Wrapf(err, "failed to set log level for %v", settingName)
+				return errors.Wrapf(err, "failed to set instance-manager log level to %v, using setting %v", setting.Value, settingName)
 			}
 		case types.SettingNameV2DataEngineLogFlags:
-			err = client.LogSetFlags(longhorn.DataEngineTypeV2, "spdk_tgt", setting.Value)
-			if err != nil {
-				return errors.Wrapf(err, "failed to set log flags for %v", settingName)
+			if types.IsDataEngineV2(im.Spec.DataEngine) {
+				err = client.LogSetFlags(longhorn.DataEngineTypeV2, "spdk_tgt", setting.Value)
+				if err != nil {
+					return errors.Wrapf(err, "failed to set spdk_tgt log flags to %v, using setting %v", setting.Value, settingName)
+				}
```

Comment:
Are we deprecating this setting?

Reply:
Yes, that's the conclusion of the discussion yesterday.  No need for granularity to set it separately for instance-manager and spdk_tgt.  That will need to be updated in the docs.

Reply:
Oh, wait.  We deprecated V2DataEngineLogLevel.  LogFlags needs to remain.  That is a separate feature of SPDK logging.